packages/trpc/server/routers/loggedInViewer/bookingForwarding.handler.ts

```diff
@@ -0,0 +1,266 @@
+    throw new Error("start_date_and_end_date_required");
+  }
+
+  // Validate start date is before end date
+  if (dayjs(input.startDate).isAfter(dayjs(input.endDate))) {
+    throw new Error("start_date_must_be_before_end_date");
+  }
+
+  // Validate start and end date are in the future
+  if (dayjs(input.startDate).isBefore(dayjs().subtract(1, "d").startOf("day"))) {
+    throw new Error("start_date_must_be_in_the_future");
+  }
+
+  let toUserId;
+  let toTeamId;
+
+  if (input.toUsernameOrEmail) {
+    const isEmail = isEmailValidator(input.toUsernameOrEmail);
+    const user = await prisma.user.findFirst({
+      where: {
+        ...(isEmail && { email: input.toUsernameOrEmail }),
+        ...(!isEmail && { username: input.toUsernameOrEmail }),
+      },
+      select: {
+        id: true,
+      },
+    });
+
+    // Validate user belongs to team if isEmail = true
+    if (isEmail) {
+      const userTeam = await prisma.membership.findFirst({
+        select: {
+          teamId: true,
+        },
+        where: {
+          userId: user?.id,
+        },
+      });
+
+      // This will only fetch userTeamId but allow org invitations if member it's on another team from org
+      if (userTeam) {
+        toTeamId = userTeam.teamId;
+      }
+    }
+
+    toUserId = user?.id;
+  }
+  if (!toUserId) {
+    throw new Error("user_not_found");
+  }
+
+  // Validate if already exists
+  const bookingForwarding = await prisma.bookingForwarding.findFirst({
+    where: {
+      start: dayjs(input.startDate).startOf("day").toISOString(),
+      end: dayjs(input.endDate).endOf("day").toISOString(),
+      userId: ctx.user.id,
+      ...(!!toTeamId && { toTeamId: Number(input.teamId) }),
+      toUserId: toUserId,
+    },
+  });
+
+  if (bookingForwarding) {
+    throw new Error("booking_forwarding_already_exists");
+  }
+
+  // Prevent infinite redirects
+  const existingBookingForwarding = await prisma.bookingForwarding.findFirst({
+    select: {
+      userId: true,
+      toUserId: true,
+      status: true,
+    },
+    where: {
+      userId: toUserId,
+      toUserId: ctx.user.id,
+    },
+  });
+
+  if (existingBookingForwarding) {
+    throw new Error("booking_forwarding_infinite_not_allowed");
+  }
+
+  // Count number of booking redirects with accepted status
+  const acceptedBookingForwardings = await prisma.bookingForwarding.count({
+    where: {
+      userId: ctx.user.id,
+      start: {
+        gte: new Date().toISOString(),
+      },
+    },
+  });
+
+  // Limit to 10 always
+  if (acceptedBookingForwardings >= 10) {
+    throw new Error("booking_redirect_limit_reached");
+  }
+
+  const createdForwarding = await prisma.bookingForwarding.create({
+    data: {
+      uuid: uuidv4(),
+      start: dayjs(input.startDate).startOf("day").toISOString(),
+      end: dayjs(input.endDate).endOf("day").toISOString(),
+      userId: ctx.user.id,
+      ...{ ...(input.teamId && { toTeamId: Number(input.teamId) }) },
```

Comment:
Do we allow passing a teamdId? 

`BookingForwardingForm` only includes  
```
{ 
  startDate: string;
  endDate: string;
  toUsernameOrEmail: string;
}
```
so I don't think we ever have a teamId coming in, maybe I am missing something

Reply:
Yeah, since there wasn't much details in the description I kinda middle ground it just in case. From peer's comment will make this only for teams and update inputs. :)